packages/@aws-cdk/core/test/resource.test.ts

```diff
@@ -318,6 +318,45 @@ describe('resource', () => {
 
   });
 
+  test('applyRemovalPolicy available for child resources', () => {
+    class Child extends Resource {
+      constructor(scope: Construct, id: string) {
+        super(scope, id);
+
+        this.node.defaultChild = new CfnResource(
+          this,
+          'Resource',
+          { type: 'ChildResourceType' },
+        );
+      }
+    }
+
+    class Parent extends Construct {
+      public readonly child: IResource
+
+      constructor(scope: Construct, id: string) {
+        super(scope, id);
+
+        this.child = new Child(this, 'Child');
+      }
+    }
+
+    const stack = new Stack();
+    const parent = new Parent(stack, 'Parent');
+
+    parent.child.applyRemovalPolicy(RemovalPolicy.RETAIN);
```

Comment:
The parent is necessary because the point is to have `applyRemovalPolicy` available resources which are typed with `IResource`. If I only instantiate the child, it's just a `Resource`, therefore it always had the `applyResourcePolicy` method attached.

Reply:
So, can we do `const child: IResource = new Child(...`? 🙂

Reply:
Thanks! That works :slightly_smiling_face: Updated